precompiles/vesting/tx.go

```diff
@@ -88,6 +90,14 @@ func (p Precompile) FundVestingAccount(
 		),
 	)
 
+	if contract.CallerAddress != origin {
```

Comment:
We moved this check on the ics20 precompile right? https://github.com/evmos/evmos/blob/d06c4c8709fdaa8bd55a983bcf524f4a50851c17/precompiles/ics20/types.go#L360

Reply:
Not quite, it uses a hardcoded msg type only working with `Transfer`. It's a good point that we should generify this even further to accept any msg type. Will update that in another PR after we release v15.